internal/app/appauth/storage.go

```diff
@@ -0,0 +1,335 @@
+package appauth
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"math/big"
+	mrand "math/rand"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/caos/oidc/pkg/oidc"
+	"github.com/caos/oidc/pkg/op"
+	"github.com/oklog/ulid"
+	"gopkg.in/square/go-jose.v2"
+)
+
+type Storage struct {
+	lock      sync.Mutex
+	appConfig *StorageConfig
+	clients   map[string]op.Client
+	requests  map[string]AuthRequest
+	keySet    jose.JSONWebKeySet
+	key       *rsa.PrivateKey
+}
+
+type StorageConfig struct {
+	Domain string `default:"http://localhost:8080"`
+	Debug  bool
+	Pkix   AuthPkixConfig
+}
+
+type AuthPkixConfig struct {
+	Organization  string
+	Country       string
+	Province      string
+	Locality      string
+	StreetAddress string
+	PostalCode    string
+}
+
+func NewAuthStorage(cfg *StorageConfig) op.Storage {
+
+	client := initLocalClient(cfg.Debug)
+
+	name := pkix.Name{
+		Organization:  []string{cfg.Pkix.Organization},
+		Country:       []string{cfg.Pkix.Country},
+		Province:      []string{cfg.Pkix.Province},
+		Locality:      []string{cfg.Pkix.Locality},
+		StreetAddress: []string{cfg.Pkix.StreetAddress},
+		PostalCode:    []string{cfg.Pkix.PostalCode},
+	}
+
+	key, keySet := initKeys(name)
+
+	return &Storage{
+		appConfig: cfg,
+		requests:  make(map[string]AuthRequest),
+		clients: map[string]op.Client{
+			client.GetID(): client,
+		},
+		key:    key,
+		keySet: keySet,
+	}
+}
+
+func initKeys(name pkix.Name) (*rsa.PrivateKey, jose.JSONWebKeySet) {
+
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		panic(err)
+	}
+
+	cert := &x509.Certificate{
+		SerialNumber: big.NewInt(1658),
```

Comment:
Set serial number that is clearly dummy number

```suggestion
		SerialNumber: big.NewInt(1),
```